pkg/ingester/checkpoint.go

```diff
@@ -0,0 +1,59 @@
+package ingester
+
+import (
+	"time"
+
+	"github.com/grafana/loki/pkg/chunkenc"
+)
+
+// The passed wireChunks slice is for re-use.
+func toWireChunks(descs []*chunkDesc, wireChunks []Chunk) ([]Chunk, error) {
+	if cap(wireChunks) < len(descs) {
+		wireChunks = make([]Chunk, len(descs))
```

Comment:
I wonder if it wouldn't be better to append what is missing. But without benchmark, I can't prove it so we shall see in the future.